src/silo_api/database_mutex.cpp

```diff
@@ -1,31 +1,24 @@
 #include "silo_api/database_mutex.h"
 
-#include <mutex>
+#include <atomic>
 #include <utility>
 
 #include "silo/database.h"
 
 namespace silo_api {
 
-silo_api::FixedDatabase::FixedDatabase(
-   const silo::Database& database,
-   std::shared_lock<std::shared_mutex>&& mutex
-)
-    : lock(std::move(mutex)),
-      database(database) {}
-
 void silo_api::DatabaseMutex::setDatabase(silo::Database&& new_database) {
-   const std::unique_lock lock(mutex);
```

Comment:
This is the line where it got stuck before. A log message before would have helped a lot: "Acquiring unique lock to switch out datasets"